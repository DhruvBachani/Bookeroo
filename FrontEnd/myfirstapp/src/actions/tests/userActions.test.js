import React from 'react';
import reducer from '../../../src/reducers'
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';

const mockStore = configureMockStore();
const store = mockStore([]);

const getAllUnapprovedUsers = () => ({
    type: 'GET_USER_REQUESTS'
});

const getAllUsers = () => ({
    type: 'GET_ALL_USERS'
});

const approveUser = () => ({
    type: 'GET_USER_REQUESTS'
});

const banUser = () => ({
    type: 'GET_USER_REQUESTS'
});

describe('personActions', () => {
    it('creates GET_USER_REQUESTS when getting all unapproved users', () => {
        store.dispatch(getAllUnapprovedUsers());
        expect(store.getActions()).toMatchSnapshot();
    })

    it('creates GET_ALL_USERS when it is called successfully', () => {
        store.dispatch(getAllUsers());
        expect(store.getActions()).toMatchSnapshot();
    })

    it('creates GET_USER_REQUESTS when approving a user', () => {
        store.dispatch(approveUser());
        expect(store.getActions()).toMatchSnapshot();
    })

    it('creates GET_USER_REQUESTS when banning a user', () => {
        store.dispatch(banUser());
        expect(store.getActions()).toMatchSnapshot();
    })
})