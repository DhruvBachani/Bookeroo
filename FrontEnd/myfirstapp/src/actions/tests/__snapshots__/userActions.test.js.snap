// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`personActions creates GET_ALL_USERS when it is called successfully 1`] = `
Array [
  Object {
    "type": "GET_USER_REQUESTS",
  },
  Object {
    "type": "GET_ALL_USERS",
  },
]
`;

exports[`personActions creates GET_USER_REQUESTS when approving a user 1`] = `
Array [
  Object {
    "type": "GET_USER_REQUESTS",
  },
  Object {
    "type": "GET_ALL_USERS",
  },
  Object {
    "type": "GET_USER_REQUESTS",
  },
]
`;

exports[`personActions creates GET_USER_REQUESTS when banning a user 1`] = `
Array [
  Object {
    "type": "GET_USER_REQUESTS",
  },
  Object {
    "type": "GET_ALL_USERS",
  },
  Object {
    "type": "GET_USER_REQUESTS",
  },
  Object {
    "type": "GET_USER_REQUESTS",
  },
]
`;

exports[`personActions creates GET_USER_REQUESTS when getting all unapproved users 1`] = `
Array [
  Object {
    "type": "GET_USER_REQUESTS",
  },
]
`;
